### DISCUSSION TITLE
Questioning the Preference: Community vs. Enterprise in Odoo 17

### DISCUSSION THREAD

**Discussion by zibbonkhadka** [OCA/odoo-community.org#181] [General]
Hello,

I find myself increasingly excited every time I delve into the Odoo 17 Enterprise edition. However, I’ve noticed that some individuals advocate for the Community edition over the Enterprise version, which puzzles me. Are there underlying reasons for this preference that I might be unaware of, beyond the advantage of complete ownership? I struggle to comprehend why the Community edition is often favored so highly in comparison to the Enterprise edition.

**Comment by rvalyi** [comment]
Hello @zibbonkhadka well it largely boils down to the general debate open source vs proprieritary software indeed. Running Odoo Enterprise will often mean running around 30% of proprieritary code, which is in general quite enough in an ERP to lock you in for the rest of your life. It will require a quite talented integrator and often 1 or 2 years if you really want to move out to the community edition later... 

For instance many companies call us after the service provided by Odoo or by some official partner don't meet their expectation... And often they have purchased a 3 years Odoo Enterprise contract that will suck enough from their ERP budget so they are completely screwed during these years. 30% of proprieritary code might not seem a lot, but it's enough to push the traditional ERP sales techniques where the customer will feel screwed (remember around 70% of ERPs project fail in general, so the traditional ERP license market is really a nasty market).

Also, year after year that is true that Odoo Enterprise does the basic things much better.

But in the OCA you will often find people doing challenging projects instead or work in countries where Odoo Enterprise fail to do the basics things (like here in Brazil).

Moreover, for people doing complex projects, extending proprieritary Odoo Enterprise modules would be an extremely brittle business: both because of legal terms and because of proprieritary code tends to be abandonned or completely changed without any warranty such as the one a powerful community like the OCA provides.

Furthermore, being guided by its company speculative valuation, Odoo will often adopt the famous "fake it until you make it" or "throw shit against the wall to see what sticks" while your project might deserve better than deception.

Finally, I would say the problem with Odoo Enterprise is more with the poor quality service that is sold together. Historically, Odoo has been pressuring its best "partners" to become mere resellers with completely bubblish sales targets. So a system called "market for lemon" as been at play in the oficial partner network (read https://saylordotorg.github.io/text_introduction-to-economic-analysis/s19-01-market-for-lemons.html ) and year after year the average level has dropped to shocking low levels (like many partners frauding even their certifications). Basically, aside from 3 or 4 skilled companies, the skilled guys moved away (to the OCA or out of Odoo). So at the end of the day, the people that will harass you to sell you some Odoo Enterprise or the guys Odoo will put you in contact with for the integration will often have very poor skills and often poor ethics as well to meet their sales target... This is exactly the opposite of what made the success of Odoo during its first decade when it was still fully open source and people weren't there only just for the easy money of the license business.
So often the service you will get along with your Odoo Enterprise will have a very low quality and true ERP projects use to fail for less than that.

That being said, if Odoo Enterprise is reputed to work well for your country and for your business, if you are a very small company with very basic requirements, then the Odoo Enterprise SaaS might still be the most affordable option indeed.

**Comment by pedrobaeza** [comment]
Continuing with the Raphaël's reasoning, the enterprise layer forces you to pay forever if you want to not only use it, but to keep the data related to them, as you are forced to uninstall the modules - deleting these data on that process - when you retire from paying. On contrary, with 100% community modules, you only pay for the services you want, but data is yours - although you have to make sure that custom developments made to you by any implementator are also open for avoiding other lock-in -.

Take also into account the asymmetry in the enterprise world, as most of the times, there's not enough with the Odoo SA modules, so or you have to pay for 2 services, the enterprise licenses for the Odoo SA code and another payment for the rest of the modules (OCA or custom ones), or restrict to only the core features, which makes Odoo far less attractive. And there's also the problem mentioned by Raphaël about finding skilled enough enterprise implementators.

I think that it's also interesting that you check this video of a talk I gave some years ago about the Odoo ecosystem, more oriented to those wanting to be implementators, but also serves for a general overview of these issues: https://www.youtube.com/watch?v=rMFDzA34Peo

**Reply by pedrobaeza** [reply]
Anyway, @zibbonkhadka which is your role? Are you an Odoo final customer or do you want to become an Odoo implementator?

**Reply by zibbonkhadka** [reply]
I am a software engineer, and we are trying to decide if Odoo Community Edition is a good fit. How robust is the OCA accounting module? Can we use it as an alternative to QuickBooks Desktop? If not, are there other stable OCA modules available to get the most out of the accounting features?

**Reply by pedrobaeza** [reply]
Yes, thousands of companies are doing their accounting with Odoo community + OCA modules, including all our customers. Install at least the OCA modules `account_financial_report`, `account_usability`, `account_reconcile_oca`, `mis_builder` + the localized reports for your country (example: `l10n_us_mis_financial_report`). And there are a lot more for special needs (financial risk, payment orders, financial budgeting, etc). Explore them in https://odoo-community.org/shop

**Reply by zibbonkhadka** [reply]
What makes OCA accounting modules different from Enterprise and 3rd party accounting modules (e.g., Odoo Mates, Cybrosys Techno Solutions)?

What do you recommend for companies that are discovering Odoo and want to move away from the proprietary world (like Intuit)? We are located in the United States and are looking for guidance with the following to get started:

1. Migration (QuickBooks Desktop to Odoo Community Edition)
2. Accounting

**Reply by rvalyi** [reply]
Odoo Enterprise brings quite a lot of value for a reasonable start price. But it's important to understand that Odoo 60% worldwide annual growth is very often accompanied with amateurish superficial features, the famous "fake it until you make it". Proprietary for proprietary, sometimes other established proprietary packages might be a better option when they have been doing exactly what you need for decades. With Odoo "there will be blood" during the implementation. If you go OCA and skilled integrator (or if you are an engineer) then you have a chance to fix what you need. If you go Enterprise or Odoo SaaS, things will be be much harder to fix if it doesn't meet your exact requirements.

It's all of a trade off. Odoo is orders of magnitude more flexible than any dinosaur ERP really. But it does comes with bugs and superficial features in the Enterprise package as well. A small company might cope with it (it has no choice after all), while a larger company may get very expensive bills if they go with Enterprise with the fake Odoo partner service that usually goes along with it and is sadly largely fake, just here to fake there is a huge skilled service offer so you can buy licenses blindly while this service offer is a big lie.

So for instance you said your are an engineer. Can you believe Odoo uses floating point IEEE 754 for all its accounting?
https://medium.com/@olivier.s/demystifying-floating-point-arithmetic-why-0-1-0-2-0-3-673b9c4bcf9a
Well, with extreme care and university math understanding, good programmers manage to get it working, but still, in every Odoo release this generate a wide variety of bugs such as: https://github.com/odoo/odoo/issues?q=is%3Aissue+is%3Aopen+rounding+

This is just an example of the gap between the marketing and the reality. Another example that I was recently confronted with: https://github.com/odoo/odoo/pull/174868

"What makes OCA accounting modules different from Enterprise and 3rd party accounting modules (e.g., Odoo Mates, Cybrosys Techno Solutions)?"

The later brand modules are a bit focused on being a branded packages to promote the service company behind them. Nothing wrong, but instead the OCA modules are more like the UNIX philosophy of having a lot of small and extremely carefully designed modules that work together to do exactly what you need. Also just like in the scientific world, the OCA is a peer reviewed environment where it's much harder for poor design to make its way in. Sure, nobody is winning a lot of money when somebody download an OCA module, so just like for tools like fffmeg or busybox there isn't a ton of aggressive marketing to convince you it's the right tool for you. But well, seasoned professionals know they are the best modules and eventually you will be just one more spreading the word.

Finally the OCA has solid processes such as Continuous Integration, Continuous Package Deployment, but also migration processes to ensure the OCA modules will be there in the next version, even if a maintainer were to move away. You will hardly get similar warranties from corporate packages...

**Reply by zibbonkhadka** [reply]
Thank you so much for the clarification.

**Comment by zibbonkhadka** [comment]
@rvalyi @pedrobaeza Thank you so much for the details.

**Comment by anajuaristi** [comment]
Hello, just to say... you don't need to decide always one or the other. We
are perfectly using enterprise with community modules. There is only a few
minimal community module quantity that are not compatible with enterprise.
Odoo core is the same in community and enterprise. Enterprise is just a
private github branch that you can add or not on your projects. To use this
branch is mandatory to pay licenses by user to Odoo. That's the difference.
If you use community and you want to go to enterprise you only need to add
the branch on your project installation and pay. If you want to go from
enterprise to community you must uninstall all enterprise modules you are
using and erase the branch from the server, so probably it will became in a
complex project.
It's quite simple. If your customer needs enterprise functionality, it's
much better to pay and use it that trying to build it yourself. In OCA
there is several modules that can cover part of enterprise functionality
but not everything . On the other side, OCA is covering a lot of
functionality that enterprise is not covering anyway. Even if year after
year Odoo core is covering more and more becaming better and better.
Depending on the customer needs, on user number, industry and so on...
enterprise + OCA community addons and even custom addons is the optimal
solution.
My2cents

El vie, 30 ago 2024 a las 8:27, Pedro M. Baeza ***@***.***>)
escribió:

> Yes, thousands of companies are doing their accounting with Odoo community
> + OCA modules, including all our customers. Install at least the OCA
> modules account_financial_report, account_usability, account_reconcile_oca,
> mis_builder + the localized reports for your country (example:
> l10n_us_mis_financial_report). And there are a lot more for special needs
> (financial risk, payment orders, financial budgeting, etc). Explore them in
> https://odoo-community.org/shop
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/OCA/discussions/181#discussioncomment-10494627>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABNRX4II347VVTE5VEIZEDZUAGFXAVCNFSM6AAAAABNJQD2WCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBZGQ3DENY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 

Ana Juaristi Olalde

Avanzosc S.L. <http://www.avanzosc.es/>
Office phone / Tfono oficina: (+34) 943 02 69 02

[image: AvanzOsc Odoo Ready Partner][image: Asociado AEODOO]

El contenido de esta comunicación y de toda su documentación anexa es
confidencial y se dirige exclusivamente a su destinatario. El uso no
autorizado de esta información está prohibido por la legislación
vigente. Si usted no es el destinatario le rogamos nos lo indique, no
comunique su contenido a terceros y proceda a su destrucción. Disculpe
las molestias que le haya ocasionado la recepción indebida de este
e-mail. Sus datos figuran en los sistemas de tratamiento de Avanzosc,
S.L., a quien usted puede dirigirse para ejercer sus derechos de
acceso, rectificación, limitación, supresión, oposición y
portabilidad, así como retirar su consentimiento y realizar cualquier
otra solicitud sobre esta materia, en calle Julio Urkijo, 34, bajo,
20720 Azkoitia (Gipuzkoa), T. 943026902 – ***@***.***

Komunikazio honen edukia eta dokumentazio erantsia konfidentziala da
eta hartzaileak bakarrik jaso beharko luke. Indarrean dagoen legeriak
debekatu egiten du bertan eskainitako informazioa baimenik gabe
erabiltzea. Komunikazioa zuri iritsi bazaizu, baina zu ez bazara
hartzailea, mesedez, guri jakinarazi, eta jasotako informazioa ez
inori jakinarazi eta suntsitu. Barkatu okerreko email hau jasotzeak
eragindako eragozpenak. Zure datuak Avanzosc, S.L. enpresaren
tratamendu-sistemetan sartuta daude. Zure datuak ikustea, zuzentzea,
ezereztea, tratamendua mugatzea, tratamenduari aurka egitea eta datuen
portabilitatea eska dezakezu. Zure baimena edonoiz ezeztatu dezakezu,
baita gai honi buruzko bestelako edozein eskaria egin ere.
Horretarako, enpresara jo dezakezu helbide honetan: Julio Urkijo
kalea, 34, behea, 20720 Azkoitia (Gipuzkoa), T. 943026902 –
***@***.***

This message and all documents attached to it are confidential and
intended only for the person or entity to which it is addressed. Any
use of this information by unauthorised persons is prohibited under
current legislation. If you received this message by error, please
advise us, destroy it and refrain from communicating its contents to
third parties. We apologise for any inconvenience receiving this email
improperly may cause to you. Your personal data are included in a
processing sistem owned by Avanzosc, S.L. If you want to exercise your
rights of access, rectification, erasure, restriction, objection and
portability, or you want to revoque your consent or make any other
inquire about this matter, you can contact Avanzosc, S.L. at Julio
Urkijo, 34, bajo, 20720 Azkoitia (Gipuzkoa), T. 943026902 –
***@***.***

**Reply by pedrobaeza** [reply]
I don't agree in this sentence: "it's much better to pay and use it that trying to build it yourself", due to these reasons (and more):

- You have to always pay, or lose the feature. If you pay it or develop it as an open source model, you can keep the module without paying.
- And not only to pay always, but to pay for every user you add in the system, so if your company scales up, or if it's of medium size, the cost starting to be very high.
- With enterprise modules, you can't develop extensions in open, so you should be restricted by your own extensions, not as OCA modules, where a lot of companies are adding features to the existing modules.

**Comment by yajo** [comment]
Hi there! The answer to your question is that you have to make the choice yourself. Some pros and cons of each option:

Using only Odoo CE + OCA modules:
- Pros:
  - Fully open source.
  - No vendor lock-in.
  - Full control over the migration process with OpenUpgrade.
- Cons: 
  - Less integrated solutions.
  - Uglier.
  - ~~No official android/ios app.~~
  - Migrations take a lot of effort.
  - You maintain things yourself (among other OCA members).
  - Contribution is often not easy.
  - You loose many fancy features from EE.

Using Odoo EE + OCA modules:
- Pros:
  - Fancy features from EE.
  - Migration is easier, faster and sooner.
  - Beautier.
  - More integrated solutions.
  - ~~Official mobile apps.~~
  - Maintenance of EE part is fully on Odoo's shoulders (although not everybody is happy with how they do it).
- Cons:
  - Vendor lock-in.
  - Closed source can cause problems with licenses, CI, maintainability, etc. (although you're free to change those sources for your customers once you're a partner, with custom patches).
  - Closed source module overrides effort isn't usually shared.
  - Migration process is a black box.
  - Costs money.
  - Contribution is often not easy.

One extra point to keep into consideration is that, for the final customer, both cost money. You either pay your OCA partner, or you pay Odoo, but you pay anyways.

Probably I'm missing some on each of them, and probably somebody will come to tell you that OCA is beautier 😄, but those should help you make your choice.

**Reply by liebana** [reply]
Sorry if I'm kind of hijacking the thread, but what are for you those "Fancy features from EE"?

**Reply by pedrobaeza** [reply]
About the pro for enterprise "Official mobile apps.", that's not true, as the app is not maintained since 2 versions, and now via PWA, everything goes directly as an app, both community and enterprise.

**Reply by yajo** [reply]
Ah true @pedrobaeza, I didn't recall that!

**Reply by yajo** [reply]
@liebana I mean almost all features from EE. You can see the Odoo Experience keynote for each version. Many features showcased there are usually EE.

**Reply by legalsylvain** [reply]
> CE : Migrations take a lot of effort.
> EE : Migration is easier, faster and sooner.

Are you sure about that topic of upgrade ? _Personnaly_, I never used EE upgrade services, so, I can not estimate if the Odoo migration process is correct.
However, I heard several integrators saying that the Odoo upgrade process was quite bad, and and even a few integrators saying that despite the EE maintenance contract, they were using openupgrade.

Is the Odoo Upgrade service now OK ? or does it depend on the complexity of the project ? (or other ?)

> CE : Uglier.
> EE : beautier.

Even with OCA/web_responsive ? I don't use EE, but AFAIK, it looks like CE + web_responsive is quite like EE. Or did I missed something ?

thanks !

**Reply by jguenat** [reply]
Hello,

I've done a few migrations with EE upgrade service and OpenUpgrade.

In my experience time required is mostly dependent of project complexity : initial state of database and volume of custom modules.

EE migration service

Pros:
- Easy to use
- Availability for latest Odoo versions

Cons:
- Blackbox (harder to customize migration process)
- Not 100% reproducible ; outcome might be different if you run migrations at different time
- You have to wait on Odoo migration support if blocking issue during the process

OpenUpgrade

Pros:
- You can easily integrate custom migration scripts in the process
- 100% reproducible (if you don't pull code in between of course)

Cons:
- Migration environment harder to setup (at least the first time, you can find help with doc/talks, once you have enough knowledge you can end up with a one line command to do your migration) 
- Latest Odoo versions not available, if you have the expertise/time you can contribute to OpenUpgrade but then yes it takes more time than EE upgrade service.

Would be interested in other feedback.

Thanks :)

**Comment by zibbonkhadka** [comment]
Thank you, everyone, for chiming in and providing great insights. Please feel free to contact me if you think you could assist with the migration process. We are primarily seeking help with the following:

- Assistance with migrating all legacy data from QuickBooks Desktop to Odoo 17 Community Edition.
- Support in configuring OCA accounting modules.

**Comment by john675645** [comment]
The choice between Odoo 17 Community and Enterprise often comes down to **features, cost, and flexibility**. The **Community edition** is free and open-source, giving businesses **full control, customization, and no licensing fees**, which appeals to startups or those with strong developer support. On the other hand, the **Enterprise edition** offers advanced features, priority support, and smoother updates, making it ideal for businesses seeking **out-of-the-box functionality and faster deployment**.

Many users prefer Community because it allows **complete ownership and flexibility**, while Enterprise is preferred for convenience and extended features. At[ Ledger Labs](https://theledgerlabs.com/), we often help businesses **assess which edition aligns with their growth needs**, ensuring they invest in the right Odoo solution without unnecessary overhead.
